tachyon/zk/plonk/examples/fibonacci/fibonacci3_circuit.h

```diff
@@ -0,0 +1,164 @@
+#ifndef TACHYON_ZK_PLONK_EXAMPLES_FIBONACCI_FIBONACCI3_CIRCUIT_H_
+#define TACHYON_ZK_PLONK_EXAMPLES_FIBONACCI_FIBONACCI3_CIRCUIT_H_
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "tachyon/zk/plonk/constraint_system/circuit.h"
+#include "tachyon/zk/plonk/examples/fibonacci/is_zero_chip.h"
+
+namespace tachyon::zk::plonk {
+
+template <typename F>
+struct Fibonacci3Config {
+  using Field = F;
+
+  Fibonacci3Config(const Selector& selector, const AdviceColumnKey& a,
+                   const AdviceColumnKey& b, const AdviceColumnKey& c,
+                   IsZeroConfig<F>&& a_equals_b, const AdviceColumnKey& output)
+      : selector(selector),
+        a(a),
+        b(b),
+        c(c),
+        a_equals_b(std::move(a_equals_b)),
+        output(output) {}
+
+  Fibonacci3Config Clone() const {
+    return Fibonacci3Config(selector, a, b, c, a_equals_b.Clone(), output);
+  }
+
+  Selector selector;
+  AdviceColumnKey a;
+  AdviceColumnKey b;
+  AdviceColumnKey c;
+  IsZeroConfig<F> a_equals_b;
+  AdviceColumnKey output;
+};
+
+template <typename F>
+class Fibonacci3Chip {
+ public:
+  explicit Fibonacci3Chip(Fibonacci3Config<F>&& config)
+      : config_(std::move(config)) {}
+
+  static Fibonacci3Config<F> Configure(ConstraintSystem<F>& meta) {
+    Selector selector = meta.CreateSimpleSelector();
+    AdviceColumnKey a = meta.CreateAdviceColumn();
+    AdviceColumnKey b = meta.CreateAdviceColumn();
+    AdviceColumnKey c = meta.CreateAdviceColumn();
+    AdviceColumnKey output = meta.CreateAdviceColumn();
+
+    AdviceColumnKey is_zero_advice_column = meta.CreateAdviceColumn();
+    IsZeroConfig<F> a_equals_b = IsZeroChip<F>::Configure(
+        meta,
+        [&selector](VirtualCells<F>& meta) {
+          return meta.QuerySelector(selector);
+        },
+        [&a, &b](VirtualCells<F>& meta) {
+          // FIXME(ashjeong): Logic is not implemented as an in-line
+          // return, as Linux/Mac systems would result in different outcomes.
+          // Further research into this phenomenon may be needed.
+          auto x = (meta.QueryAdvice(a, Rotation::Cur()));
+          auto y = (meta.QueryAdvice(b, Rotation::Cur()));
```

Comment:
Please avoid using `auto` as much as possible!

```suggestion
          std::unique_ptr<Expression<F>> x = meta.QueryAdvice(a, Rotation::Cur());
          [auto](std::unique_ptr<Expression<F>>) y = meta.QueryAdvice(b, Rotation::Cur());
```